tests/component/component.cpp

```diff
@@ -499,3 +493,29 @@ TEST(Component, constructors)
     libcellml::ComponentPtr c3 = std::move(c2);
     EXPECT_EQ("my_name", c3->name());
 }
+
+TEST(Component, multiParentWithAddComponentBugIssue399)
+{
+    libcellml::ModelPtr model = std::make_shared<libcellml::Model>();
+    libcellml::ComponentPtr parent = std::make_shared<libcellml::Component>();
+    parent->setName("parent_component");
+    libcellml::ComponentPtr child1 = std::make_shared<libcellml::Component>();
+    child1->setName("child1");
+    libcellml::ComponentPtr child2 = std::make_shared<libcellml::Component>();
+    child2->setName("child2");
+    libcellml::ComponentPtr child3 = std::make_shared<libcellml::Component>();
+    child3->setName("child3");
+
+    parent->addComponent(child1);
+    parent->addComponent(child2);
+    parent->addComponent(child3);
+    model->addComponent(parent);
+
+    EXPECT_EQ(size_t(3), parent->componentCount());
+    EXPECT_EQ(size_t(0), child2->componentCount());
+
+    child2->addComponent(child3);
```

Comment:
Do we test anywhere what happens if a component is added as a child of itself?
`c->addComponent(c); ` ?

Reply:
Not sure easy enough to add that test, I would expect that to return `false`.

Reply:
Thanks